### ISSUE TITLE
Celery workers stop receiving new tasks after "Couldn't ack, reason:time out"

### ISSUE THREAD

**Issue by Glavin001** [celery/celery#2415]
I am hoping someone can help me with this. I have been trying to figure out what is going wrong to no avail. I want to continue using Celery, however if I cannot resolve this issue then I will have to try and find another solution as I require low maintenance and high uptime (always running and being used every second, and I will not always be available to restart it if an error occurs). I want any errors to fail silently (with error message but not crash) and reconnect to the server, if previously unavailable (however RabbitMQ should always be running, so it should always be available).

## Software
- Windows 7 Professional
- RabbitMQ 3.2.4 with Erlang R16B03
- Celery:

```
C:\cygwin\home\nevops\opc-client>python -m celery report

software -> celery:3.1.16 (Cipater) kombu:3.0.23 py:2.7.7
            billiard:3.3.0.18 py-amqp:1.4.6
platform -> system:Windows arch:32bit, WindowsPE imp:CPython
loader   -> celery.loaders.default.Loader
settings -> transport:amqp results:disabled

CELERY_ACKS_LATE: False
BROKER_URL: 'amqp://guest:********@localhost:5672//'
CELERY_ENABLE_UTC: True
CELERY_DISABLE_RATE_LIMITS: True
CELERY_ACCEPT_CONTENT: ['json']
CELERY_IMPORTS:
    ('tasks',)
BROKER_HEARTBEAT: 2
BROKER_CONNECTION_RETRY: True
CELERY_ROUTES: {
    'tasks._PollTagsTask': {   'queue': 'poll_queue'},
    'tasks._ShipReadingsTask': {   'queue': 'ship_queue'}}
BROKER_CONNECTION_MAX_RETRIES: 0
CELERY_TASK_SERIALIZER: 'json'
CELERY_RESULT_SERIALIZER: 'json'
```

## Bug / Issue

- Workers for queue intermittently stop running, although there are still tasks being queued that should be processed.
- This occurs in both of my Tasks, `PollTagsTask` and `ShipReadingsTask`. Sometimes one or the other, and sometimes both stop.
- If I simply manually restart the worker then it reconnects and continues as usual. If it errors, I want it to try and reconnect autonomously

### Logs

Here is a sample of the logs when it occurs:

```
[2014-11-21 12:51:25,045: INFO/MainProcess] Received task: tasks._ShipReadingsTa
sk[6db9d1a1-dfc0-43d1-96c7-4d26f529125a]
[[2014-11-21 12:51:25,075: INFO/MainProcess] Received task: tasks._ShipReadingsT
ask[85d25f77-6f5e-40e1-90c5-96810ab3ca5a]
[2014-11-21 12:51:25,154: INFO/MainProcess] Task tasks._ShipReadingsTask[2515a56
f-bc34-4cca-bdd2-fb69316797ae] succeeded in 0.108999967575s: None
[2014-11-21 12:51:25,186: CRITICAL/MainProcess] Couldn't ack 584715, reason:time
out('timed out',)
Traceback (most recent call last):
File "C:\Python27\lib\site-packages\kombu\message.py", line 93, in ack_log_err
or
self.ack()
File "C:\Python27\lib\site-packages\kombu\message.py", line 88, in ack
self.channel.basic_ack(self.delivery_tag)
File "C:\Python27\lib\site-packages\amqp\channel.py", line 1583, in basic_ack
self._send_method((60, 80), args)
File "C:\Python27\lib\site-packages\amqp\abstract_channel.py", line 56, in _se
nd_method
self.channel_id, method_sig, args, content,
File "C:\Python27\lib\site-packages\amqp\method_framing.py", line 221, in writ
e_method
write_frame(1, channel, payload)
File "C:\Python27\lib\site-packages\amqp\transport.py", line 182, in write_fra
me
frame_type, channel, size, payload, 0xce,
File "C:\Python27\lib\socket.py", line 224, in meth
return getattr(self._sock,name)(*args)
timeout: timed out
[2014-11-21 12:51:25,638: INFO/MainProcess] Task tasks._ShipReadingsTask[6db9d1a
1-dfc0-43d1-96c7-4d26f529125a] succeeded in 0.576999902725s: None
[2014-11-21 12:51:25,654: INFO/MainProcess] Task tasks._ShipReadingsTask[85d25f7
7-6f5e-40e1-90c5-96810ab3ca5a] succeeded in 0.515000104904s: None
```

## Desired Goal

- I can accept lost task messages (many happen every every second so loss is minimal)
- I want it to *continue* with the next task if any one errors

Thank you in advance for any help!

**Comment by starplanet** [comment]
I have this problem too. I am using celery to realize a ETL system, and I need run celery on windows to load data into client database which is installed on windows server.
How can I fix this problem? If you can give me some suggestions, I will be appreciated for that very much.

**Comment by ask** [comment]
Closing this, as we don't have the resources to complete this task.

May be fixed in master, let's see if comes back after 4.0 release.

**Comment by Saw-mon-and-Natalie** [comment]
Still experiencing this issue on **Windows 7**

```
software -> celery:4.3.0 (rhubarb) kombu:4.6.3 py:3.6.2
            billiard:3.6.0.0 py-amqp:2.5.0
platform -> system:Windows arch:32bit, WindowsPE
            kernel version:7 imp:CPython
loader   -> celery.loaders.default.Loader
settings -> transport:amqp results:disabled
```

**Comment by joagonzalez** [comment]
Any news? experiencing same issue on Windows Server 2019 with this setup:

```
celery==4.4.0
broker = rabbit
```

```
OS Name             : Microsoft Windows Server 2019 Standard
OS Version          : 10.0.17763 N/A Build 17763
OS Manufacturer     : Microsoft Corporation
OS Configuration    : Standalone Server
OS Build Type       : Multiprocessor Free
System Boot Time    : 5/13/2020, 3:37:38 PM
System Manufacturer : VMware, Inc.
System Model        : VMware Virtual Platform
System Type         : x64-based PC
System Directory    : C:\Windows\system32
System Locale       : en-us;English (United States)
Hotfix(s)           : 3 Hotfix(s) Installed.,[01]: KB4514366,[02]: KB4512577,[03]: KB4512578
```


ERROR
`[2020-06-17 14:13:40,256: CRITICAL/MainProcess] Couldn't ack 28, reason:error(10054, 'An existing connection was forcibly closed by the remote host')
Traceback (most recent call last):
  File "c:\python27\lib\site-packages\kombu\message.py", line 131, in ack_log_error
    self.ack(multiple=multiple)
  File "c:\python27\lib\site-packages\kombu\message.py", line 126, in ack
    self.channel.basic_ack(self.delivery_tag, multiple=multiple)
  File "c:\python27\lib\site-packages\amqp\channel.py", line 1394, in basic_ack
    spec.Basic.Ack, argsig, (delivery_tag, multiple),
  File "c:\python27\lib\site-packages\amqp\abstract_channel.py", line 59, in send_method
    conn.frame_writer(1, self.channel_id, sig, args, content)
  File "c:\python27\lib\site-packages\amqp\method_framing.py", line 172, in write_frame`

**Comment by auvipy** [comment]
can you try celery==4.4.5?

**Comment by joagonzalez** [comment]
Yes, I am going to test with 4.4.5 and I'll be back with feedback. Thanks for the answer

**Comment by joagonzalez** [comment]
I've been testing it for a few weeks and the problem was solved after upgrade to celery=4.4.5. Tanks for the answer @auvipy
